src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -4021,6 +4006,19 @@ public override BoundNode VisitAsOperator(BoundAsOperator node)
             return result;
         }
 
+        public override BoundNode VisitSuppressNullableWarningExpression(BoundSuppressNullableWarningExpression node)
+        {
+            var result = base.VisitSuppressNullableWarningExpression(node);
+
+            Debug.Assert(!IsConditionalState);
```

Comment:
>Debug [](start = 12, length = 5)

Assert would fail if I write `if (c!) {}`

Reply:
`IsConditionalState` is always `true` currently because `base.VisitSuppressNullableWarningExpression()` calls `VisitRvalue(node.Expression)`. For an expression such as `(b && F(out var o))!? o : null)`, the condition should not be treated as an rvalue. Added `SuppressNullableWarning_Condition` unit test for now.